components/org.wso2.carbon.identity.conditional.auth.functions.analytics/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/analytics/internal/AnalyticsFunctionsServiceComponent.java

```diff
@@ -170,6 +208,29 @@ protected void unsetServerConfigurationService(ServerConfigurationService server
         AnalyticsFunctionsServiceHolder.getInstance().setServerConfigurationService(null);
     }
 
+    @Reference(
+            name = "server.configuration.service",
+            service = ServerConfigurationService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetServerConfigurationService"
+    )
+    protected void setServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+
+        if (LOG.isDebugEnabled()) {
```

Comment:
\`\`\`suggestion
            LOG.debug("Setting the serverConfigurationService");
            LOG.info("Server configuration service set in AnalyticsFunctionsServiceHolder");
\`\`\`